File: src/main/java/com/cuupa/classificator/to/DocumentBuilder.java
package com.cuupa.classificator.to;

import java.io.IOException;
import java.io.InputStream;

public class DocumentBuilder {

	private final InputStream inputStream;
	private final String filename;
	private final int fileSize;

	private DocumentBuilder(InputStream inputStream, String filename, int fileSize) {
		this.inputStream = inputStream;
		this.filename = filename;
		this.fileSize = fileSize;
	}

	public static DocumentBuilder create(InputStream inputStream, String fileName, int fileSize) {
		return new DocumentBuilder(inputStream, fileName, fileSize);
	}

	public Document build() throws IOException {
		Document document = new Document();
		document.setFileName(filename);
		byte[] content = new byte[fileSize];
		inputStream.read(content);
		document.setContent(content);
		return document;
	}

}

File: src/main/java/com/cuupa/classificator/to/DocumentBuilder.kt
package com.cuupa.classificator.to

import java.io.InputStream

class DocumentBuilder private constructor(private val inputStream: InputStream, private val filename: String,
                                          private val fileSize: Int) {
    fun build(): Document {
        val content = ByteArray(fileSize)
        inputStream.read(content)
        return Document(content, filename, false)
    }

    companion object {
        fun create(inputStream: InputStream, fileName: String,
                   fileSize: Int): DocumentBuilder {
            return DocumentBuilder(inputStream, fileName, fileSize)
        }
    }

}